tools/clang/test/HLSLFileCheck/hlsl/template/DependentTypeInObjectParamInvalid.hlsl

```diff
@@ -0,0 +1,21 @@
+// RUN: %dxc -E main -T vs_6_6 -HV 2021 -enable-16bit-types -ast-dump %s | FileCheck %s
+
+// CHECK: error: 'MyStruct' cannot be used as a type parameter where a scalar is required
+// CHECK-NEXT: Texture2D<vector<T, 4> > tex = ResourceDescriptorHeap[0];
+
+// CHECK: note: in instantiation of function template specialization 'foo<MyStruct>' requested here
+// CHECK-NEXT: return foo<MyStruct>().f;
+
+template<typename T>
+T foo() {
+  Texture2D<vector<T, 4> > tex = ResourceDescriptorHeap[0];
```

Comment:
Because struct cannot be used in a vector that way.  This is to check that the skip for dependent type doesn't prevent this from being checked once we have a concrete type.

I couldn't do this check the same way with the previous version of the function that took a texture object parameter because you would get an error on constructing the texture object when attempting to supply the parameter.  Since then I found a better way to test this without the dynamic resource, that's closer to the positive case but with the invalid type.  I'll update the test with that.

Reply:
I changed the test (moved to VerifierTest as well).  Hopefully that alleviates any concerns.